tests/test_cloudshell.py

```diff
@@ -0,0 +1,21 @@
+import unittest
+from msal.cloudshell import _scope_to_resource
+
+class TestScopeToResource(unittest.TestCase):
+
+    def test_expected_behaviors(self):
+        for scope, expected_resource in {
+            "https://analysis.windows.net/powerbi/api/foo":
+                "https://analysis.windows.net/powerbi/api",  # A special case
+            "https://pas.windows.net/CheckMyAccess/Linux/.default":
+                "https://pas.windows.net/CheckMyAccess/Linux/.default",  # Special case
+            "https://double-slash.com//scope": "https://double-slash.com/",
+            "https://single-slash.com/scope": "https://single-slash.com",
+            "guid/some/scope": "guid",
+            "797f4846-ba00-4fd7-ba43-dac1f8f63013/.default":  # Realistic GUID
+                "797f4846-ba00-4fd7-ba43-dac1f8f63013"
```

Comment:
https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/785#discussion_r1929651794 is my fault. Even though ARM's app ID can be used to get an access token locally, it fails in Cloud Shell:

> Audience 797f4846-ba00-4fd7-ba43-dac1f8f63013 is not a supported MSI token audience.

`6dae42f8-4368-4678-94ff-3960e28e3630` (`https://aks-aad-server.azure.com`) works, but `e406a681-f3d4-42a8-90b6-c2b029497af1` (`https://storage.azure.com`) results in error

> Timeout waiting for token from portal. Audience: e406a681-f3d4-42a8-90b6-c2b029497af1

Reply:
```suggestion
            "6dae42f8-4368-4678-94ff-3960e28e3630/.default":  # Realistic GUID
                "6dae42f8-4368-4678-94ff-3960e28e3630"
```

Reply:
The current implementation actually works for this kind of pattern regardless of its guid. But, OK, the PR has now been changed to use that real GUID.